datastream-server/src/test/java/com/linkedin/datastream/server/TestDatastreamTask.java

```diff
@@ -38,47 +38,84 @@
 public class TestDatastreamTask {
 
   @Test
-  public void testAcquireWithDependencies() throws Exception {
+  public void testAcquireWithDependencies() {
     Datastream stream = DatastreamTestUtils.createDatastream("dummy", "dummy", "dummy");
     stream.getMetadata().put(DatastreamMetadataConstants.TASK_PREFIX, DatastreamTaskImpl.getTaskPrefix(stream));
 
     DatastreamTaskImpl task = new DatastreamTaskImpl(Collections.singletonList(stream));
     ZkAdapter mockZkAdapter = mock(ZkAdapter.class);
     task.setZkAdapter(mockZkAdapter);
 
-    task.addDependency("task0");
+    DatastreamTaskImpl task0 = new DatastreamTaskImpl(Collections.singletonList(stream));
+    ZkAdapter mockZkAdapter2 = mock(ZkAdapter.class);
+    task0.setZkAdapter(mockZkAdapter2);
+    when(mockZkAdapter2.checkIsTaskLocked(anyString(), anyString(), anyString())).thenReturn(true);
+
+    task.addDependency(task0);
     task.acquire(Duration.ofMillis(60));
     verify(mockZkAdapter, atLeastOnce()).waitForDependencies(any(DatastreamTaskImpl.class), any(Duration.class));
   }
 
   @Test(expectedExceptions = DatastreamTransientException.class)
-  public void testCreateNewTaskFromUnlockedTask() throws Exception {
+  public void testCreateNewTaskFromUnlockedTask() {
```

Comment:
Nit: Not your fault, Do you think this test can be changed to validate the exception using AssertThrows to ensure that all the blocks including the new code is getting executed.

Reply:
done